### PR TITLE
fix: (Store) Fix entity format

### PR DESCRIPTION
**Pull request by dimamarksman**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#5217

#### Please provide a brief summary of this pull request.
Fix for Entity Format. It should give the ability access entity properties directly e.g. `myEntity.entityProperty`

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts [hunk 1]
```diff
@@ -25,15 +21,18 @@ interface ReqDTO {
...
 }
 
-class Money extends BaseValue<MoneyDTO> {
-    constructor(value?: MoneyDTO) {
-        super(value);
+class Money extends BaseValue<MoneyDTO> implements MoneyDTO {
+    amount: number;
+    currency: string;
+
+    constructor(dto: MoneyDTO) {
```
Comment (fkolar) [inline libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts:28]: how are. you using DTO here? I dont think its used at all right ?

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts [hunk 2]
```diff
@@ -48,100 +47,137 @@ interface ReqDTO {
...
     amount: number;
+    total: Money;
     lineItems: Array<LineItem>;
 }
 
 class Requisition extends BaseEntity<ReqDTO> {
     uniqueName: IdentityKey;
     title: string;
     amount: number;
+    total: Money;
```
Comment (fkolar) [inline libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts:58]: I am not sure how this this Requisition work and how we are using Parent BaseEntity

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts [hunk 3]
```diff
@@ -1,46 +1,69 @@
...
+            // otherwise redirect to the default value.
+            // Need it to make working entity.getter flow
+            if (propertyKey in value) {
+                return Reflect.get(value, propertyKey, value);
+            }
         }
-
-        return Reflect.get(target as any, prop);
+        // default flow
+        return Reflect.get(target, propertyKey, receiver);
```
Comment (fkolar) [inline libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts:31]: Did not we discuss that we will not use Reflect.***

File: libs/store/src/lib/infrastructure/persistence/store/entity-store.spec.ts [hunk 4]
```diff
@@ -2,18 +2,23 @@ import { of } from 'rxjs';
...
-class User extends BaseEntity<{}>{
-    constructor(public id: IdentityKey, public name: string, public age: number) {
-        super({ id, name, age });
+class UserDTO {
+    id: IdentityKey;
+    name: string;
+    age: number;
+}
+class User extends BaseEntity<UserDTO> {
+    constructor(dto: UserDTO) {
```
Comment (fkolar) [inline libs/store/src/lib/infrastructure/persistence/store/entity-store.spec.ts:16]: These examples are not aligned, somewhere we are constructor and accessing .value and somewhere not.

**Review by fkolar — CHANGES_REQUESTED** [review]

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts (hunk 3 above)
Comment (fkolar) [reply libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts:31]: Yes. But are you using we need it ? With you previous asnwer ?

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts]: I removed this constructor here so it's not required

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts (hunk 2 above)
Comment (dimamarksman) [reply libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts:50]: I left the core things the same, so it uses the same approach as previously. 
A proxied entity relies on internal `composite.value` namespace.
`Proxy.get` trap tries to find the property in the `composite.value` namespace, if it’s not there then redirect to the default flow.
`Proxy.set` trap tries to set a new value in the `composite.value` namespace if possible, if not redirect to the default assignment process

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts (hunk 3 above)
Comment (dimamarksman) [reply libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts:31]: sorry didn't get your last question

File: libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/store/src/lib/infrastructure/persistence/domain/state-handler.spec.ts]: Parent class constructor is used by default

**Review by genereg1 — APPROVED** [review]

**Review by fkolar — APPROVED** [review]

**Pull request merged**